synchronizer/synchronizer.go

```diff
@@ -192,6 +205,11 @@ func (s *Synchronizer) OnLocalTimeout() {
 	s.consensus.StopVoting(s.currentView)
 
 	s.configuration.Timeout(timeoutMsg)
+
+	if s.currentView < s.highQC.View()+hotstuff.View(s.pipelinedViews) {
```

Comment:
This construction:
```go
s.currentView < s.highQC.View()+hotstuff.View(s.pipelinedViews)
```
is used three times in this file. And could beneficially be done with a method:
```go
func (s Synchronizer) canAdvance(view hotstuff.View) bool {
  return view < s.highQC.View()+s.pipelinedViews
}
```
I think the compiler will most likely inline this code anyway, so shouldn't give any added overheads.

Reply:
@meling For me it seems a bit unlogical that you can add two views. 
Do you think, it would be better to keep `pipelinedViews` an `uint` and then define an add function on the view type?

Reply:
Done in [eb95aa1](https://github.com/relab/hotstuff/pull/78/commits/eb95aa1b566321e76dec2296f05e3403fe58758e)

Reply:
> @meling For me it seems a bit unlogical that you can add two views.
> Do you think, it would be better to keep `pipelinedViews` an `uint` and then define an add function on the view type?

Using `uint` is fine, if we use a func to “hide” the type cast… it was anyway being casted before added to a view…

Reply:
I've now reviewed the changes in more detail. I think it is fine to use `uint8` since it would essentially only need to cast it to `hotstuff.View` one additional place (in the `isInPipeline...()`), which would be fine since it wouldn't be part of code people read. So feel free to revert it to `uint8`.